test/server/HttpTaskServerTest.java

```diff
@@ -0,0 +1,269 @@
+        int idTask = task1.getId();
+        int idSubtask = subtask3.getId();
+        String task1Json = httpUtils.getGson().toJson(manager.getTaskById(idTask));
+        String subtaskJson = httpUtils.getGson().toJson(manager.getSubtaskById(idSubtask));
+
+        HttpResponse<String> task1Response = httpUtils.sendPOSTRequest("http://localhost:8080/tasks" + idTask,
+                task1Json);
+        HttpResponse<String> subtask3Response = httpUtils.sendPOSTRequest("http://localhost:8080/subtasks"
+                + idSubtask, subtaskJson);
+
+        assertEquals(201, task1Response.statusCode(), "код ответа для Task не совпадает");
+        assertNotEquals(unexpectedTask, manager.getTaskById(idTask).toString(), "Задачи Task совпадают, " +
+                "обновление задачи не произошло");
+        assertEquals(201, subtask3Response.statusCode(), "код ответа для Subtask не совпадает");
+        assertNotEquals(unexpectedSubtask, manager.getSubtaskById(idSubtask).toString(), "Задачи Subtask " +
+                "совпадают, обновление задачи не произошло");
+    }
+
+    @Test
+    public void checkingThatTheTaskIsNotAddedIfTheTimeIntervalsMatch() throws IOException, InterruptedException {
+        manager.addEpic(epic5);
+        manager.addTask(task1);
+        manager.addSubtask(subtask3);
+        task2.setStartTime(task1.getStartTime().plusMinutes(2));
+        subtask4.setStartTime(subtask3.getStartTime().plusMinutes(2));
+        String task2Json = httpUtils.getGson().toJson(task2);
+        String subtask4Json = httpUtils.getGson().toJson(subtask4);
+
+        HttpResponse<String> task2Response = httpUtils.sendPOSTRequest("http://localhost:8080/tasks", task2Json);
+        HttpResponse<String> subtask4Response = httpUtils.sendPOSTRequest("http://localhost:8080/subtasks",
+                subtask4Json);
+
+        assertEquals(406, task2Response.statusCode(), "Код ответа задания Task не совпадает");
+        assertEquals(1, manager.getListFromHashTask().size(), "Некоректное колличество задач Task");
+        assertEquals(406, subtask4Response.statusCode(), "Код ответа задания Subtask не совпадает");
+        assertEquals(1, manager.getListFromHashSubtask().size(), "Некоректное колличество задач " +
+                "Subtask");
+    }
+
+    @Test
+    public void checkingTheReceiptOfTheTaskListOverTheNetwork() throws IOException, InterruptedException {
+        manager.addEpic(epic5);
+        manager.addTask(task1);
+        manager.addTask(task2);
+        manager.addSubtask(subtask3);
+        manager.addSubtask(subtask4);
+
+        HttpResponse<String> responseTask = httpUtils.sendGETRequest("http://localhost:8080/tasks/");
+        HttpResponse<String> responseSubtask = httpUtils.sendGETRequest("http://localhost:8080/subtasks/");
+        HttpResponse<String> responseEpic = httpUtils.sendGETRequest("http://localhost:8080/epics/");
+
+        List<Task> actualListTask = httpUtils.getGson().fromJson(responseTask.body(),
```

Comment:
Соответственно тут нужно будет заменить на 
```suggestion
                List<Task> actualListTask = httpUtils.getGson().fromJson(
                responseTask.body(),
                new TypeToken<List<Task>>() {}.getType()
        );
        List<Subtask> actualListSubtask = httpUtils.getGson().fromJson(
                responseSubtask.body(),
                new TypeToken<List<Subtask>>() {}.getType()
        );
        List<Epic> actualListEpic = httpUtils.getGson().fromJson(
                responseEpic.body(),
                new TypeToken<List<Epic>>() {}.getType()
        );
```